### DISCUSSION TITLE
Allow 'where' class restrictions on the Generic Parameters of Generic Classes

### DISCUSSION THREAD

**Discussion by Joseph-Melberg** [dotnet/csharplang#4286] [General]
Given
`public class A : B<ImplParam> {}`
`public class ImpleParam : Param {}`
`public abstract class Param {}`
`public abstract class B<T> where T : Param {}`

I would like to be able to restrict the method

`public void Foo<TClass>() `
` let TParameter : Param`
` where TClass : B<TParameter>`
`{}`
  

The current work around is 

`public void Foo<TClass,TParameter>() where TClass : B<TParameter> where TParameter : Param {}`

Would it be possible to incorporate this feature?

**Comment by huoyaoyuan** [comment]
This could be a part of HKT(#339).

**Reply by 333fred** [reply]
Yeah, this strikes me as a higher-kinded type request.
